xprom/xpromkafka/collector.go

```diff
@@ -0,0 +1,230 @@
+package xpromkafka
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/gojekfarm/xtools/xkafka"
+	"github.com/gojekfarm/xtools/xprom/semconv"
+)
+
+var (
+	defaultLatencyBuckets = []float64{
+		0.001, // 1ms
+		0.002, // 2ms
+		0.005, // 5ms
+		0.01,  // 10ms
+		0.02,  // 20ms
+		0.05,  // 50ms
+		0.1,   // 100ms
+		0.2,   // 200ms
+		0.5,   // 500ms
+		1,     // 1s
+		2,     // 2s
+		5,     // 5s
+		10,    // 10s
+	}
+)
+
+// Collector provides metrics for xkafka.Producer and xkafka.Consumer.
+type Collector struct {
```

Comment:
Why can't these be otel based Counter, Histograms, etc?

I have used them in [courier-go](https://github.com/gojek/courier-go/blob/1a9171ad1ec083b23ad800e8271841491022b74a/otelcourier/metric.go) and it works with the `prometheus.DefaultRegistry` as well.

Reply:
This might be helpful during otel migration. For now I want to keep xprom purely for prometheus

Reply:
Got it!